From 28c8e0755b91ae4a8628bfcc1b1ca39b4c494bc5 Mon Sep 17 00:00:00 2001
From: Hugues Kamba-Mpiana <hugues.kambampiana@arm.com>
Date: Tue, 19 Dec 2023 13:17:15 +0000
Subject: [PATCH] toolchain: Fix Arm GNU Toolchain 13.2 Rel1 compatibility

Arm GNU Toolchain 13.2 Rel1 no longer includes <cstdint> transitively.
Include `cstdint` explicitely where needed.

Signed-off-by: Hugues Kamba-Mpiana <hugues.kambampiana@arm.com>
---
 source/application/api/common/include/ClassificationResult.hpp | 1 +
 source/profiler/include/Profiler.hpp                           | 1 +
 source/use_case/asr/include/UseCaseHandler.hpp                 | 2 ++
 source/use_case/kws/include/UseCaseHandler.hpp                 | 2 ++
 4 files changed, 6 insertions(+)

diff --git a/source/application/api/common/include/ClassificationResult.hpp b/source/application/api/common/include/ClassificationResult.hpp
index 244849f..b3b3f76 100644
--- a/source/application/api/common/include/ClassificationResult.hpp
+++ b/source/application/api/common/include/ClassificationResult.hpp
@@ -18,6 +18,7 @@
 #define CLASSIFICATION_RESULT_HPP
 
 #include <string>
+#include <cstdint>
 
 namespace arm {
 namespace app {
diff --git a/source/profiler/include/Profiler.hpp b/source/profiler/include/Profiler.hpp
index 8ae9e15..07e0de4 100644
--- a/source/profiler/include/Profiler.hpp
+++ b/source/profiler/include/Profiler.hpp
@@ -19,6 +19,7 @@
 
 #include "hal.h"
 
+#include <cstdint>
 #include <string>
 #include <map>
 #include <vector>
diff --git a/source/use_case/asr/include/UseCaseHandler.hpp b/source/use_case/asr/include/UseCaseHandler.hpp
index 60e27fc..04e6199 100644
--- a/source/use_case/asr/include/UseCaseHandler.hpp
+++ b/source/use_case/asr/include/UseCaseHandler.hpp
@@ -19,6 +19,8 @@
 
 #include "AppContext.hpp"
 
+#include <cstdint>
+
 namespace arm {
 namespace app {
 
diff --git a/source/use_case/kws/include/UseCaseHandler.hpp b/source/use_case/kws/include/UseCaseHandler.hpp
index 279acf0..eb28a8f 100644
--- a/source/use_case/kws/include/UseCaseHandler.hpp
+++ b/source/use_case/kws/include/UseCaseHandler.hpp
@@ -19,6 +19,8 @@
 
 #include "AppContext.hpp"
 
+#include <cstdint>
+
 namespace arm {
 namespace app {
 
-- 
2.34.1

